### ISSUE TITLE
Token not returning

### ISSUE THREAD

**Issue by starlisa** [XeroAPI/xero-netstandard-oauth2-starter-dotnet-core#9]
I deployed the app to an Azure host: https://xeronetstandardapp20230522130045.azurewebsites.net using the deploy tool in VisualStudio 2019.

I modified the appsettings.json to include the ClientId and ClientSecret, and set the CallbackURI to https://xeronetstandardapp20230522130045.azurewebsites.net/Home which I also set as a redirect URI on the configuration page for the KCHS app: https://developer.xero.com/app/manage/app/5162436b-8445-4aa9-8c2e-a5605519665f/configuration

When I run the app, I get the Home page. I click on Connect to Xero, and I get the authorize.xero.com page. When I click on Allow Access, I'm redirected back to the Home page. But without a token, so it again only shows the Connect to Xero button.

I haven't changed anything else in the project.  Why would it fail to return a token?

**Comment by starlisa** [comment]
On the chance that the CallbackURI was supposed to be https://xeronetstandardapp20230522130045.azurewebsites.net/Authorization/Callback rather than https://xeronetstandardapp20230522130045.azurewebsites.net/Home, I tried that, and got the following errors:

AggregateException: One or more errors occurred. (Parameter is required (Parameter 'refresh_token'))
System.Threading.Tasks.Task.ThrowIfExceptional(bool includeTaskCanceledExceptions)

ArgumentException: Parameter is required (Parameter 'refresh_token')
IdentityModel.Client.Parameters.AddRequired(string key, string value, bool allowDuplicates, bool allowEmptyValue)

**Comment by MattJamesChampion** [comment]
I've run into this same issue and believe I've got a fix.

In `LocalStorageTokenIO.cs` (specifically the `StoreToken` function), there's a check to see if `xerotoken.json` exists and, if so, writes the token details to it. Crucially, the issue is that by default this file _doesn't_ exist and there's no branch that attempts to create it.

Thankfully, `File.WriteAllText` actually attempts to create the file if it doesn't already exist, bypassing the need for the `if` statement in the first place. As such, removing the `File.Exists` check and simply serializing and storing the token works as expected.

I've just created a pull request which resolves this problem.
